src/zks_wallet/wallet.rs

```diff
@@ -255,28 +281,191 @@ where
+        let era_provider = self.get_era_provider()?;
+
+        let gas_limit: U256 = {
+            let address_str = format!("{l1_token_address:?}");
+            let default_erc20_deposit_gas_limit = 300000_u64; // FIXME make it a constant.
+            let is_mainnet = self.get_era_provider()?.get_chainid().await? == 324_i32.into();
+            if is_mainnet {
+                (*ERC20_DEPOSIT_GAS_LIMITS)
+                    .get(&address_str)
+                    .unwrap_or(&default_erc20_deposit_gas_limit)
+                    .to_owned() // FIXME fix unwrap
+            } else {
+                default_erc20_deposit_gas_limit
+            }
+        }
+        .into();
+
+        // If the user has already provided max_fee_per_gas or gas_price, we will use
+        // it to calculate the base cost for the transaction
+        let gas_price = if let Some(max_fee_per_gas) = max_fee_per_gas {
+            max_fee_per_gas
+        } else if let Some(gas_price) = gas_price {
+            gas_price
+        } else {
+            era_provider.get_gas_price().await?
+        };
+
+        let l2_gas_limit = U256::from(3_000_000_u32);
+
+        let base_cost: U256 = self
+            .get_base_cost(
                 l2_gas_limit,
-                gas_per_pubdata_byte,
-                Default::default(),
-                refund_recipient,
+                DEPOSIT_GAS_PER_PUBDATA_LIMIT.into(),
                 gas_price,
-                gas_limit,
-                l1_value,
             )
             .await?;
 
-        Ok(receipt)
+        // ERC20 token, `msg.value` is used only for the fee.
+        let value = base_cost + operator_tip;
+
+        let data: Bytes = {
+            let bridge_contract = abi::l1_bridge_contract();
+
+            #[allow(clippy::expect_used)]
+            let contract_function = bridge_contract
+                .function("deposit")
+                .expect("failed to get deposit function parameters");
+
+            let params = (
+                to,
+                l1_token_address,
+                amount,
+                l2_gas_limit,
+                U256::from(DEPOSIT_GAS_PER_PUBDATA_LIMIT),
+            );
+
+            #[allow(clippy::expect_used)]
+            contract_function
+                .encode_input(&params.into_tokens())
+                .expect("failed to encode deposit function parameters")
+                .into()
+        };
+
+        let chain_id = eth_provider.get_chainid().await?.as_u64();
+
+        println!("bridge_address: {bridge_address:?}");
+        let bridge_address: Address = match bridge_address {
+            Some(address) => address,
+            None => {
+                let bridge_contracts = era_provider.get_bridge_contracts().await?;
+                bridge_contracts.l1_erc20_default_bridge
+            }
+        };
+        println!("bridge_address: {bridge_address:?}");
+
+        // FIXME where do I set the nonce?
```

Comment:
The nonce is automatically set when the transaction is sent but maybe is worth checking that out

Reply:
You are right. Solved @ 41afa0e